### PR TITLE
Raven db 9141

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: src/Raven.Client/ServerWide/DatabaseRecord.cs [hunk 1]
```diff
@@ -147,6 +148,19 @@ public void DeletePeriodicBackupConfiguration(long backupTaskId)
             }
         }
 
+        public void EnsureTaskNameIsNotUsed(string taskName)
+        {
+
+            if (string.IsNullOrEmpty(taskName))
+                throw new ArgumentException("Can't validate task's name because the provided task name name is null or empty.");
```
Comment (ml054) [inline src/Raven.Client/ServerWide/DatabaseRecord.cs]: name name

**Review by ml054 — COMMENTED** [review]
General note: Could you unify casing in words:
ETL, SQL, external replication?

File: src/Raven.Client/ServerWide/DatabaseRecord.cs [hunk 2]
```diff
@@ -147,6 +148,19 @@ public void DeletePeriodicBackupConfiguration(long backupTaskId)
...
+
+            if (ExternalReplications.Any(x => x.Name == taskName))
+                throw new InvalidOperationException($"Can't use task name {taskName} there is already an external replications task with that name");
+            if (RavenEtls.Any(x => x.Name == taskName))
+                throw new InvalidOperationException($"Can't use task name {taskName} there is already an etl task with that name");
+            if (PeriodicBackups.Any(x => x.Name == taskName))
+                throw new InvalidOperationException($"Can't use task name {taskName} there is already a backup task with that name");
```
Comment (Danielle9897) [inline src/Raven.Client/ServerWide/DatabaseRecord.cs]: external replications -> **External Replication**
etl -> **ETL**
backup -> **Backup**

File: src/Raven.Client/ServerWide/ETL/RavenEtlConfiguration.cs [hunk 3]
```diff
@@ -28,6 +28,11 @@ public override bool UsingEncryptedCommunicationChannel()
             return true;
         }
 
+        public override string GetDefaultTaskName()
+        {
+            return $"Raven etl to {ConnectionStringName}";
```
Comment (Danielle9897) [inline src/Raven.Client/ServerWide/ETL/RavenEtlConfiguration.cs]: Raven etl -> **RavenDB ETL**

File: src/Raven.Client/ServerWide/ETL/SqlEtlConfiguration.cs [hunk 4]
```diff
@@ -86,6 +86,11 @@ public override bool UsingEncryptedCommunicationChannel()
             }
         }
 
+        public override string GetDefaultTaskName()
+        {
+            return  $"Sql Etl to {ConnectionStringName}";
```
Comment (Danielle9897) [inline src/Raven.Client/ServerWide/ETL/SqlEtlConfiguration.cs]: Sql Etl -> **SQL ETL**

File: src/Raven.Client/ServerWide/ExternalReplication.cs [hunk 5]
```diff
@@ -88,5 +88,10 @@ public string GetMentorNode()
         {
             return MentorNode;
         }
+
+        public string GetDefaultTaskName()
+        {
+            return $"External replication to {ConnectionString}";
```
Comment (Danielle9897) [inline src/Raven.Client/ServerWide/ExternalReplication.cs]: External replication -> **External replication**

File: src/Raven.Client/ServerWide/PeriodicBackup/PeriodicBackupConfiguration.cs [hunk 6]
```diff
@@ -54,6 +54,11 @@ public string GetMentorNode()
             return MentorNode;
         }
 
+        public string GetDefaultTaskName()
+        {
+            return $"Periodic backup, full:{FullBackupFrequency} incremental:{IncrementalBackupFrequency}";
```
Comment (Danielle9897) [inline src/Raven.Client/ServerWide/PeriodicBackup/PeriodicBackupConfiguration.cs]: backup -> **Backup**

File: src/Raven.Client/ServerWide/DatabaseRecord.cs [hunk 7]
```diff
@@ -147,6 +148,19 @@ public void DeletePeriodicBackupConfiguration(long backupTaskId)
...
 
+        public void EnsureTaskNameIsNotUsed(string taskName)
+        {
+
+            if (string.IsNullOrEmpty(taskName))
+                throw new ArgumentException("Can't validate task's name because the provided task name is null or empty.");
+
+            if (ExternalReplications.Any(x => x.Name == taskName))
+                throw new InvalidOperationException($"Can't use task name {taskName} there is already an External Replications task with that name");
+            if (RavenEtls.Any(x => x.Name == taskName))
```
Comment (ayende) [inline src/Raven.Client/ServerWide/DatabaseRecord.cs:159]: Aren't we missing SQL ETL tasks here too?

Comment (ayende) [reply src/Raven.Client/ServerWide/DatabaseRecord.cs:159]: And subscriptions?

File: src/Raven.Server/ServerWide/Commands/UpdateExternalReplicationCommand.cs [hunk 8]
```diff
@@ -19,6 +19,11 @@ public UpdateExternalReplicationCommand(string databaseName) : base(databaseName
 
         public override string UpdateDatabaseRecord(DatabaseRecord record, long etag)
         {
+            if (string.IsNullOrEmpty(Watcher.Name))
+            {
+                Watcher.Name = Watcher.GetDefaultTaskName();
+            }
+            record.EnsureTaskNameIsNotUsed(Watcher.Name);
```
Comment (ayende) [inline src/Raven.Server/ServerWide/Commands/UpdateExternalReplicationCommand.cs:26]: Shouldn't this be called for subscriptions?

**Pull request merged**